sdk/lib/opentelemetry/sdk/trace/tracer.rb

```diff
@@ -56,7 +54,8 @@ def internal_create_span(result, name, kind, trace_id, span_id, parent_span_id,
             attributes = attributes&.merge(result.attributes) || result.attributes
             active_trace_config = OpenTelemetry.tracer_provider.active_trace_config
             active_span_processor = OpenTelemetry.tracer_provider.active_span_processor
-            Span.new(context, name, kind, parent_span_id, active_trace_config, active_span_processor, attributes, links, start_timestamp || Time.now, @resource, @instrumentation_library)
+            resource = OpenTelemetry.tracer_provider.resource
```

Comment:
Is there any reason not to assign this as an instance variable in the constructor?

Reply:
Nope no reason not to.  Updated.